src/cursor.rs

```diff
@@ -564,6 +597,19 @@ impl SyntaxNode {
         unsafe { self.ptr.as_ref() }
     }
 
+    #[inline]
+    fn can_take_ptr(&self) -> bool {
+        self.data().rc.get() == 1 && !self.data().mutable
+    }
+
+    #[inline]
+    fn take_ptr(self) -> ptr::NonNull<NodeData> {
+        assert!(self.can_take_ptr());
+        // increment the refcount before self gets dropped which will decrement the refcount again
+        self.data().inc_rc();
```

Comment:
Couldn't we just `mem::forget(self)` instead then?

Reply:
done, TIL